### ISSUE TITLE
Questions on Structural Coverage & Logical Coverage

### ISSUE THREAD

**Issue by retroinspect** [swtv-kaist/cs458-fall22#28]
1. Do edge-pair coverage cover 0, 1, 2 length paths? 
If so, are nodes & edges part of edge-pair test requirements?
If not, why does edge-pair test coverage subsume edge coverage from the hierarchy graph?

![image](https://user-images.githubusercontent.com/48230029/206708262-d9f01f36-b606-4fd6-b5a5-6c0ffca501ae.png)

2. <del>For Inactive Clause Coverage, is there any feasible set of test cases for `P = (A && B) || C` ?
It seems no combination can make C and P independent. <del>
I understand that it might be infeasible to meet inactive clause coverage.

3. From the definition of Restricted Inactive Clause Coverage on the below slide, Should the values chosen for minor clauses be all same for (1),(2),(3),(4) OR same for (1), (2) and (3), (4)? 

![image](https://user-images.githubusercontent.com/48230029/206721980-a884ff2d-fea6-43cc-b29f-a4e9661e3c9f.png)

**Comment by moonzoo** [comment]
Glad to see that you are studying CS458 lectures hard ;-) 

> Do edge-pair coverage cover 0, 1, 2 length paths?

No (the definition in the lecture slides are slightly different from the textbook)

> why does edge-pair test coverage subsume edge coverage from the hierarchy graph?

Since all edge-pairs in G contain all edges in G (except a graph that has an inextensible path of length 1 )

**Comment by retroinspect** [comment]
In this example, edge pair set has only [0, 1, 2] and not covers edge [0, 2] I guess.

<img width="200" alt="image" src="https://user-images.githubusercontent.com/48230029/206710851-9c91f8ad-4cd9-4481-ba1e-ded009dabbd4.png">

**Comment by moonzoo** [comment]
You are right.  Thank you for pointing out the glitch in the edge-pair definition in the lecture slides.
(I slightly modified the previous comment )
To handle that glitch, the textbook defines TR of EPC to contain each path of length up to 2, which sounds confusing.
So, in spite of the technical glitch, the lecture slides defines TR of EPC to contain every path of length 2.
